tests/test_primitives.py

```diff
@@ -0,0 +1,247 @@
+                name="table",
+                category="breakfast_table",
+                model="rjgmmy",
+                scale=[0.3, 0.3, 0.3]
+            ),
+        "position": [-0.7, -2.0, 0.2],
+        "orientation": [0, 0, 0, 1]
+    }
+    objects.append(obj_1)
+
+    primitives = [StarterSemanticActionPrimitiveSet.NAVIGATE_TO]
+    primitives_args = [(obj_1['object'],)]    
+
+    assert primitive_tester(categories, objects, primitives, primitives_args)
+
+def test_grasp():
+    categories = ["floors", "ceilings", "walls", "coffee_table"]
+
+    objects = []
+    obj_1 = {
+        "object": DatasetObject(
+                name="table",
+                category="breakfast_table",
+                model="rjgmmy",
+                scale=[0.3, 0.3, 0.3]
+            ),
+        "position": [-0.7, 0.5, 0.2],
+        "orientation": [0, 0, 0, 1]
+    }
+    obj_2 = {
+        "object": DatasetObject(
+            name="cologne",
+            category="cologne",
+            model="lyipur",
+            scale=[0.01, 0.01, 0.01]
+        ),
+        "position": [-0.3, -0.8, 0.5],
+        "orientation": [0, 0, 0, 1]
+    }
+    objects.append(obj_1)
+    objects.append(obj_2)
+
+    primitives = [StarterSemanticActionPrimitiveSet.GRASP]
+    primitives_args = [(obj_2['object'],)]    
+
+    assert primitive_tester(categories, objects, primitives, primitives_args)
+
+def test_place():
+    categories = ["floors", "ceilings", "walls", "coffee_table"]
+
+    objects = []
+    obj_1 = {
+        "object": DatasetObject(
+                name="table",
+                category="breakfast_table",
+                model="rjgmmy",
+                scale=[0.3, 0.3, 0.3]
+            ),
+        "position": [-0.7, 0.5, 0.2],
+        "orientation": [0, 0, 0, 1]
+    }
+    obj_2 = {
+        "object": DatasetObject(
+            name="cologne",
+            category="cologne",
+            model="lyipur",
+            scale=[0.01, 0.01, 0.01]
+        ),
+        "position": [-0.3, -0.8, 0.5],
+        "orientation": [0, 0, 0, 1]
+    }
+    objects.append(obj_1)
+    objects.append(obj_2)
+
+    primitives = [StarterSemanticActionPrimitiveSet.GRASP, StarterSemanticActionPrimitiveSet.PLACE_ON_TOP]
+    primitives_args = [(obj_2['object'],), (obj_1['object'],)]    
+
+    assert primitive_tester(categories, objects, primitives, primitives_args)
+
+def test_open_prismatic():
+    categories = ["floors"]
+
+    objects = []
+    obj_1 = {
+        "object": DatasetObject(
+            name="bottom_cabinet",
+            category="bottom_cabinet",
+            model="bamfsz",
+            scale=[0.7, 0.7, 0.7]
+        ),
+        "position": [-1.2, -0.4, 0.5],
+        "orientation": [0, 0, 0, 1]
+    }
+    objects.append(obj_1)
+
+    primitives = [StarterSemanticActionPrimitiveSet.OPEN]
+    primitives_args = [(obj_1['object'],)]    
+
+    assert primitive_tester(categories, objects, primitives, primitives_args)
+
+def test_open_revolute():
```

Comment:
this is awesome!! do we have tests for close prismatic / revolute?

Reply:
We'll do a better job of these when we have these properly implemented in a later version.